File: 240922/배열 놀이/play-with-array.py
a, b = map(int, input().split())
nums = list(map(int, input().split()))
for tc in range(a) :
    check = list(map(int, input().split()))
    if check[0] == 1:
        print(nums[check[1]-1])
    elif check[0] == 2:
        for i in range(a):
            if nums[i] == check[1] :
                print(i+1)
                break
        else :
            print(0)
    else :
        for i in range(check[1]-1, check[2]):
            print(nums[i], end = ' ')
        print()

File: 240922/배열 놀이/play-with-array.cpp
#include <iostream>
using namespace std;

int main() {
    // 여기에 코드를 작성해주세요.
    int a, b;
    cin >> a >> b;
    int nums[a] = {};
    for (int i = 0; i < a; i ++) {
        cin >> nums[i];
    }
    int check[b] = {};
    for (int j = 0; j < b; j ++) {
        cin >> check[j];
        if (check[0] == 1) {
            cout << nums[check[1] -1];
        }
        else if (check[0] == 2) {
            for (int k = 0; k < a; k ++) {
                if (nums[k] == check[1]) {
                    cout << i+1;
                    break;
                }
            }
            else {
                cout << 0;
            }
        }
        else {
            for (int i = check[1]-1; i < check[2]; i ++) {
                cout << nums[i] << " " << endl;
            }
        }
    }
    
    return 0;
}